packages/rocketchat-livechat/imports/server/rest/messages.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'underscore';
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+function checkAuthentication(request) {
+	const receivedToken = request.headers['x-rocketchat-livechat-token'];
+	const secretToken = RocketChat.settings.get('Livechat_secret_token');
+	return receivedToken && receivedToken === secretToken;
+}
+
+RocketChat.API.v1.addRoute('livechat/status/:visitorToken', {
+	get() {
+		if (!checkAuthentication(this.request)) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		const initialData = Meteor.call('livechat:getInitialData', this.urlParams.visitorToken);
+		const info = _.pick(initialData, 'visitor', 'room', 'departments', 'online', 'agentData');
```

Comment:
Why do you need these fields? When I see the route's name I imagine it would return only the visitor's status.

Reply:
Hi @rodrigok, so, in the beginning, the requirement was to know if there is some agent available (online) and then take some actions, but during the implementation, we realized that other information would be important too, such as "room", "departments" (to direct the visitor to the right chat), and "agentData" (to inform the visitor the name of agent). In the end, we left the route endpoint only named as status. Do you think is better create another endpoint to this extra information or just renaming the endpoint from status to something wider meaning is enough?

Reply:
Hi @rodrigok,
I think of two ways to improve this

### Proposal 1
is it better if we break that into two endpoints?

Visitor Endpoint: "http://host/api/v1/livechat/visitor/:visitorToken"

* visitor data
* visitor room
* agent serving visitor

Livechat Status: "http://host/api/v1/livechat/status"

* agents online
* available departments

### Proposal 2
or maybe break into four endpoints:

Visitor Endpoint: "http://host/api/v1/livechat/visitor/:visitorToken"

* visitor data

Rooms Endpoint: "http://host/api/v1/livechat/visitor/:visitorToken/room"

* opened rooms for the visitor or all rooms for visitor
* with this endpoint we can check who is serving the visitor

Departments Endpoint: "http://host/api/v1/livechat/department"

* list all departments
* with this endpoint we can determine and show only available departments to the visitor (we can filter for enabled departments and with agents)

Agents Endpoint: "http://host/api/v1/livechat/agent"

* list all agents
* with this endpoint we can check for online agents (we can count online agents) and get agent details to show whos serving the visitor

Department endpoint already exists, but authentication is not based on the webhook token, its based on user login. We will need to deal with this.

What do you think? Is it better?

Reply:
Hi @hmagarotto I prefer the second option with multiple endpoints with more clear results